### PR TITLE
Implement refactoring of custom direct methodology

### PR DESCRIPTION
**Pull request by riemanli**
## TL;DR
This pull request is the implementation of [this API change]([url](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/pull/178)). 

## What changed
- Modification of the "customDirectMethodology" property in the "EdpSimulator" class, changing the "variance" property to "scalar" and setting its value to 0.0.
- Function to convert a CustomDirectMethodology object to an InternalCustomDirectMethodology object.
- Addition of a new message called "CustomDirectMethodology" in the internal "direct_computation.proto" file, including a "FrequencyVariances" message with two maps for storing variances of frequency distributions and a "variance" field to store the variance of the result computed from the custom direct methodology.
- Test cases in the "MetricsServiceTest.kt" file, checking the presence of a specific metric in a list and checking if an exception is thrown when the variance in a custom methodology is not set.

### CODE REVIEW

**Comment by wfa-reviewable** [comment]
This change is [Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1265)

**Review by tristanvuong2021 — APPROVED** [review]
> Reviewed 2 of 5 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1265)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @stevenwarejones)

**Review by stevenwarejones — CHANGES_REQUESTED** [review]
> Reviewed 2 of 5 files at r1, 3 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1265)* status: all files reviewed, 1 unresolved discussion (waiting on @riemanli)

___
*[`src/main/kotlin/org/wfanet/measurement/reporting/service/api/v2alpha/ProtoConversions.kt` line 831 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1265#-Ng-LKMCBiI8vPhiQz7R:-Ng-LKMCBiI8vPhiQz7S:b-yrwjva) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/75e2ccd49db47cf7dc879f201b04c20c82190ce3/src/main/kotlin/org/wfanet/measurement/reporting/service/api/v2alpha/ProtoConversions.kt#L831)):*
> ```Kotlin
>       }
>       CustomDirectMethodology.VarianceCase.VARIANCE_NOT_SET -> {
>         error("Variance in CustomDirectMethodology is not set.")
> ```

should this really throw an error? Shouldn't the partial results still be revealed to the advertiser?

**Comment by riemanli** [comment]
*[`src/main/kotlin/org/wfanet/measurement/reporting/service/api/v2alpha/ProtoConversions.kt` line 831 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1265#-Ng-LKMCBiI8vPhiQz7R:-Ng-fstZDTHca7elOn6a:b-dla5zp) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/75e2ccd49db47cf7dc879f201b04c20c82190ce3/src/main/kotlin/org/wfanet/measurement/reporting/service/api/v2alpha/ProtoConversions.kt#L831)):*
<details><summary><i>Previously, stevenwarejones (Steven Ware Jones) wrote…</i></summary><blockquote>

should this really throw an error? Shouldn't the partial results still be revealed to the advertiser?
</blockquote></details>

The user can choose not to set `CustomDirectMethodology` during the transition. But if `CustomDirectMethodology` is set, its required fields should be specified.

**Review by stevenwarejones — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1265)* status: all files reviewed, 1 unresolved discussion (waiting on @riemanli)

___
*[`src/main/kotlin/org/wfanet/measurement/reporting/service/api/v2alpha/ProtoConversions.kt` line 831 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1265#-Ng-LKMCBiI8vPhiQz7R:-Ng-rz9UD6_IX3rhpy-a:b9bfmdy) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/75e2ccd49db47cf7dc879f201b04c20c82190ce3/src/main/kotlin/org/wfanet/measurement/reporting/service/api/v2alpha/ProtoConversions.kt#L831)):*
<details><summary><i>Previously, riemanli (Rieman) wrote…</i></summary><blockquote>

The user can choose not to set `CustomDirectMethodology` during the transition. But if `CustomDirectMethodology` is set, its required fields should be specified.
</blockquote></details>

i guess its okay since the oneof field that includes `CustomDirectMetholody` is not required.
